src/features/contact/contactverzoek/formulier/ContactverzoekFormulier.vue

```diff
@@ -472,21 +471,55 @@ watch(
 
 //////////////////////////////////////////////////////
 
+const hasContact = computed(
+  () =>
+    !!form.value.telefoonnummer1 ||
+    !!form.value.telefoonnummer2 ||
+    !!form.value.emailadres,
+);
+
+const noContactMessage =
+  "Vul minimaal een telefoonnummer of een e-mailadres in";
+
 watch(
-  [
-    telEl,
-    () =>
-      !!form.value.telefoonnummer1 ||
-      !!form.value.telefoonnummer2 ||
-      !!form.value.emailadres,
-  ],
-  ([el, hasContact]) => {
-    if (!el) return;
+  [telEl, hasContact],
+  ([el, bool]) => el && el.setCustomValidity(!bool ? noContactMessage : ""),
+);
+
+// klantinteracties/api/v1/schema/#tag/digitale-adressen/operation/digitaleadressenCreate
+const TELEFOON_PATTERN =
+  /^(0[8-9]00[0-9]{4,7}|0[1-9][0-9]{8}|\+[0-9]{9,20}|1400|140[0-9]{2,3})$/;
+
+const handleTelefoonInput = (event: Event) => {
+  const el = event.target as HTMLInputElement;
+
+  setActive();
+
+  if (!el.value || TELEFOON_PATTERN.test(el.value)) {
+    // telEl: back to custom noContactMessage if applicable, otherwise clear
     el.setCustomValidity(
-      hasContact ? "" : "Vul minimaal een telefoonnummer of een e-mailadres in",
+      el === telEl.value && !hasContact.value ? noContactMessage : "",
     );
-  },
-);
+  } else {
+    el.setCustomValidity("Vul een geldig telefoonnummer in.");
+  }
+};
+
+// https://github.com/django/django/blob/4.2/django/core/validators.py#L174.Voor
+const EMAIL_PATTERN =
+  /^[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+(\.[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)*@([a-zA-Z0-9-]+(\.[a-zA-Z0-9-]{2,63})+|\[[a-fA-F0-9:.]+\])$/;
```

Comment:
ik zie deze niet terug in de django file. heb je zelf de user en domain regexes gecombineerd?

Reply:
ja, comment toegevoegd

Reply:
tests voor telefoon en email validatie toegevoegd